docs/bsp.md

```diff
@@ -652,32 +658,38 @@ trait TestParams {
 Response:
 
 * result: `TestReport`, defined as follows
+* error: JSON-RPC code and message set in case an exception happens during the request.
 
 ```scala
-trait TestReport {
-  def items: List[TestReportItem]
-}
-trait TestReportItem {
+trait TestResult {}
+```
+
+Notification:
 
-  /** The compile times before executing tests if the target.
-    * An empty field means the target may have already
-    * been compiled beforehand. */
-  def compileReport: Option[CompileReportItem]
+* method: `buildTarget/testReport`
+* params: `TestReport` defined as follows:
 
+```scala
+trait TestReport {
+  /** The build target that was compiled. */
+  def target: BuildTargetIdentifier
+  
   /** The total number of successful tests. */
   def passed: Int
 
   /** The total number of failed tests. */
   def failed: Int
 
-  /** The total number of milliseconds it took to run the tests.
-    * Should not include compile times. */
+  /** The total number of milliseconds it took to only run the tests. */
```

Comment:
I think the previous docstring was clearer, it's good to explicitly say this should not include compile or packaging times.